pkg/apply/apply.go

```diff
@@ -872,15 +903,35 @@ func (t *TopicApplier) updatePlacementHelper(
 		desiredAssignments,
 		batchSize,
 		newTopic,
+		changes,
 	)
 }
 
+func (changes *UpdateChangesTracker) mergeReplicaAssignments(
```

Comment:
Please move the UpdateChangesTracker if it starts having methods, which likely it should at this point as the structure of the replica assignments is more complicated than simply tracking parameters changes.